dashboard/app/controllers/project_commits_controller.rb

```diff
@@ -4,12 +4,32 @@ class ProjectCommitsController < ApplicationController
   # POST /project_commits
   def create
     _, project_id = storage_decrypt_channel_id(params[:storage_id])
-    project_commit = ProjectCommit.new(project_id: project_id, object_version_id: params[:version_id], comment: params[:comment])
-    if project_commit.save
-      return head :ok
+    # Use find_or_initialize_by to detect if this is a duplicate submission.
+    project_commit = ProjectCommit.find_or_initialize_by(
+      project_id: project_id,
+      object_version_id: params[:version_id]
+    )
+
+    if project_commit.persisted?
+      # This is a duplicate submission - a comment for this project version already exists.
+      # Notify Honeybadger about the duplicate submission and do not overwrite the existing comment.
+      Honeybadger.notify(
```

Comment:
Should we return an error status code if this happens?

Reply:
Nice, I think the update you have here makes sense!

I think it's helpful to think about how this works independently from the specific use case we have currently. ie, if we ever use this controller action in another context, I think it would have been confusing to return 200 OK and not actually update the comment with the value that a client had provided.